ray-operator/controllers/ray/common/ingress.go

```diff
@@ -75,8 +75,28 @@ func BuildIngressForHeadService(cluster rayiov1alpha1.RayCluster) (*networkingv1
 	if !ok {
 		logrus.Warn(fmt.Sprintf("ingress class annotation is not set for cluster %s/%s", cluster.Namespace, cluster.Name))
 	} else {
+		// TODO: in AWS EKS, set up IngressClassName will cause an error due to conflict with annotation.
 		ingress.Spec.IngressClassName = &ingressClassName
 	}
 
 	return ingress, nil
 }
+
+// BuildIngressForRayService Builds the ingress for head service dashboard for RayService.
+// This is used to expose dashboard for external traffic.
+// RayService controller updates the ingress whenever a new RayCluster serves the traffic.
+func BuildIngressForRayService(service rayiov1alpha1.RayService, cluster rayiov1alpha1.RayCluster) (*networkingv1.Ingress, error) {
```

Comment:
do we need an ingress here? Just a Service resource is useful enough to call the port both internally and externally. We should expect users to build their own ingress right?

Reply:
I think the idea was to match the behavior of the RayCluster controller, which does have the ability to make an ingress.

Reply:
Right now it is the same as RayCluster as Dmitri said.